### PR TITLE
style(Button): update typings and propTypes usage

### PR DESCRIPTION
**Pull request by layershifter**
This PR is part of work for removing propTypes in production bundle (#524, #731).
Also, cleanups and updates typings for #1072.

### CODE REVIEW

File: src/elements/Button/Button.js [hunk 1]
```diff
@@ -16,37 +16,14 @@ import {
...
-      'linkedin',
-      'instagram',
-      'youtube',
-    ],
-    floated: SUI.FLOATS,
-    labelPosition: ['right', 'left'],
-    size: SUI.SIZES,
-  },
-}
-
```
Comment (layershifter) [inline src/elements/Button/Button.js:47]: It's gone 😄

File: src/elements/Button/Button.js [hunk 2]
```diff
@@ -16,37 +16,14 @@ import {
...
 /**
- * A Button indicates a possible user action
+ * A Button indicates a possible user action.
```
Comment (layershifter) [inline src/elements/Button/Button.js:26]: Dots.

File: src/elements/Button/Button.js [hunk 3]
```diff
@@ -240,22 +228,21 @@ class Button extends Component {
...
     )
     const rest = getUnhandledProps(Button, this.props)
     const ElementType = getElementType(Button, this.props, () => {
       if (!_.isNil(label) || !_.isNil(attached)) return 'div'
     })
-
-    let computedTabIndex
-    if (!_.isNil(tabIndex)) computedTabIndex = tabIndex
-    else if (disabled) computedTabIndex = -1
-    else if (ElementType === 'div') computedTabIndex = 0
```
Comment (layershifter) [inline src/elements/Button/Button.js:252]: Replaced with method `computeTabIndex`, looks much cleaner to me.

File: src/elements/Button/index.d.ts [hunk 4]
```diff
@@ -133,30 +142,59 @@ interface GroupProps {
...
+  /** Groups can be formatted to appear on dark backgrounds. */
+  inverted?: boolean;
 
-  /** Groups can be formatted as labeled icon buttons */
+  /** Groups can be formatted as labeled icon buttons. */
   labeled?: boolean;
 
-  /** Groups can have different sizes */
+  /** Groups can hint towards a negative consequence. */
+  negative?: boolean;
```
Comment (layershifter) [inline src/elements/Button/index.d.ts:164]: `Button.Group` typings had missing props.

File: src/index.d.ts [hunk 5]
```diff
@@ -2,8 +2,6 @@ export type InputType = 'color' | 'date' | 'datetime' | 'datetime-local' | 'emai
 
 export type SemanticCOLORS = 'red' | 'orange' | 'yellow' | 'olive' | 'green' | 'teal' | 'blue' | 'violet' | 'purple' | 'pink' | 'brown' | 'grey' | 'black' | 'twitter' | 'google plus' | 'facebook';
```
Comment (levithomason) [inline src/index.d.ts]: Looks like a bug here.  Only the true color values are available in every component (red, orange, etc).  The social colors are only available in buttons.

**Review by levithomason — CHANGES_REQUESTED** [review]

File: src/index.d.ts (hunk 5 above)
Comment (layershifter) [reply src/index.d.ts]: Indeed, I've updated type.

**Review by levithomason — APPROVED** [review]

File: src/elements/Button/Button.js (hunk 1 above)
Comment (levithomason) [reply src/elements/Button/Button.js:47]: 🎉

**Pull request merged**